hera_qm/xrfi.py

```diff
@@ -1192,28 +1380,127 @@ def xrfi_run(ocalfits_file, acalfits_file, model_file, data_file, history,
+    nsig_t : float, optional
+        The number of sigma above which to flag integrations. Default is 5.0.
+    cal_ext : str, optional
+        The extension to replace penultimate extension in a calfits file for output
+        calibration including flags. Defaults is "flagged_abs". For example, an
+        input_cal of "foo.goo.calfits" would result in "foo.flagged_abs.calfits".
+    clobber : bool, optional
+        If True, overwrite existing files. Default is False.
+
+    Returns
+    -------
+    None
+
+    """
+    history = 'Flagging command: "' + history + '", Using ' + hera_qm_version_str
+    data_files = sorted(data_files)
+    xrfi_dirs = [resolve_xrfi_path('', dfile, jd_subdir=True) for dfile in data_files]
+    basename = '.'.join(os.path.basename(data_files[0]).split('.')[0:2])
+    outdir = resolve_xrfi_path('', data_files[0])
+    # Set up extensions to find the many files
+    types = ['og', 'ox', 'ag', 'ax', 'v', 'data', 'chi_sq_renormed', 'combined']
+    mexts = ['og_metrics', 'ox_metrics', 'ag_metrics', 'ax_metrics',
+             'v_metrics', 'data_metrics', 'chi_sq_renormed', 'combined_metrics']
+    # Read in the metrics objects
+    filled_metrics = []
+    for ext in mexts:
+        if ext != 'data_metrics':
+            # Fill in 2nd metrics with 1st metrics where 2nd are not available.
+            files1 = [glob.glob(d + '/*' + ext + '1.h5')[0] for d in xrfi_dirs]
+            files2 = [glob.glob(d + '/*' + ext + '2.h5')[0] for d in xrfi_dirs]
+            uvf1 = UVFlag(files1)
+            uvf2 = UVFlag(files2)
+            uvf2.metric_array = np.where(np.isinf(uvf2.metric_array), uvf1.metric_array,
+                                         uvf2.metric_array)
+            filled_metrics.append(uvf2)
+        else:
+            # Data was only run in second iteration
+            files = [glob.glob(d + '/*' + ext + '2.h5')[0] for d in xrfi_dirs]
+            filled_metrics.append(UVFlag(files))
+
+    # Threshold each metric and save flag object
+    uvf_total = filled_metrics[0].copy()
+    uvf_total.to_flag()
+    for i, uvf_m in enumerate(filled_metrics):
+        if types[i] == 'chi_sq_renormed':
+            detrend = False
+        else:
+            detrend = True
+        uvf_f = threshold_wf(uvf_m, nsig_f=nsig_f, nsig_t=nsig_t, detrend=detrend)
+        outfile = '.'.join([basename, types[i] + '_threshold_flags.h5'])
+        outpath = os.path.join(outdir, outfile)
+        uvf_f.write(outpath, clobber=clobber)
+        uvf_total |= uvf_f
+
+    # Read non thresholded flags and combine
+    files = [glob.glob(d + '/*.flags2.h5')[0] for d in xrfi_dirs]
+    uvf_total |= UVFlag(files)
+
+    # Apply to abs calfits
+    uvc_a = UVCal()
+    # This is a terrible hack we need to fix when we have a select method.
+    uvf_file = uvf_total.copy()
+    for dfile in data_files:
+        basename = qm_utils.strip_extension(dfile)
+        abs_in = '.'.join([basename, 'abs', 'calfits'])
```

Comment:
ugh, yeah. I wanted to cut down on the options available for such a finely tuned wrapper that's meant to be run in a specific environment. I actually meant to hard code `flagged_abs`, but held off while I was testing, and wanted to run with a couple outputs. Are you OK with me hardcoding both `flagged_abs` and `abs`, rather than making them both options?

Reply:
Yes, I'm fine with that as long as it's clear in the docstring.